### PR TITLE
FW-751. Nordic nRF52840 port

### PR DESCRIPTION
**Pull request by tharczos**
Adds support for Nordic nRF52840 (Developer Kit and Dongle).

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -309,7 +311,112 @@ command_line_vars.AddVariables(
         command_line_options['msf_adapting_to_traffic'][1],# default
         validate_option,                                   # validator
         int,                                               # converter
-    ),
+    ),
```
Comment (changtengfei) [inline SConstruct:314]: This changes are redundant. The options are already defined at beginning.

**Review by changtengfei — COMMENTED** [review]
Hi @tharczos thanks for the effort on porting OpenWSN on your board. I saw lots of file from the SDK folder are added into the bsp. Are they all are used? I assume there is quite number of files are not needed for the port. Could you remove them and keep a minimal set of files living in the folder which is only for the porting?

Another thing is how we test it? is that possible to send us some demo board to verify the port works properly?

**Comment by dragondgold** [comment]
Any news on this? I'm really looking forward to an nRF52840 port of OpenWSN. I'm willing to do the necessary changes to be able to merge and test this properly.

**Comment by changtengfei** [comment]
@dragondgold , I haven't heard news from @tharczos about the update on this port. As you can see the comments above some changes are required and we also need to find a way to test the port.

**Comment by dragondgold** [comment]
@changtengfei You need 2 nRF52840 dongles or development kits to be able to test this right?

**Comment by changtengfei** [comment]
@dragondgold . I don't know. This supposes to be the implementer of the port to answer, what is required for doing the test and how.

**Comment by changtengfei** [comment]
Can one of the admins verify this patch?

File: projects/common/01bsp_radio/01bsp_radio.c [hunk 2]
```diff
@@ -109,6 +109,7 @@ int mote_main(void) {
    app_vars.flags |= APP_FLAG_TIMER;
    
    while (1) {
+
```
Comment (changtengfei) [inline projects/common/01bsp_radio/01bsp_radio.c:112]: The changes in this file should not be committed.

**Comment by changtengfei** [comment]
This PR looks pretty much ready to be merged. Only small things are required to be fix. @dragondgold , do you have sometime to resolve them?

**Comment by dragondgold** [comment]
@changtengfei I will take a look at this on the weekend! I never used OpenWSN because my project ended up dying but I have 3 nRF52840 boards in case testing is needed on real hardware.

**Comment by changtengfei** [comment]
@dragondgold that would be great!  We will buy several kits to test the code when it's ready!

**Comment by dragondgold** [comment]
Hi @changtengfei, I made a fork of @tharczos's fork and fixed the merge conflicts, should I create a PR so we continue from my fork? This is the fork: https://github.com/dragondgold/openwsn-fw_FW-751

**Comment by changtengfei** [comment]
@dragondgold thanks a lot for taking over this work! I have tried to contact with @tharczos through his email registered on OpenWSN but the email address does not exist anymore.  So I believe fork the code and continuous work on it make perfect sense. 

Yes, Please create another PR for the nRF52840 port.  Have you tried the code after fixing the conflicts on your side?  Does the network form, such as with two boards? We have the boards in the lab. Once you confirm that, I will test it on my side.

**Comment by dragondgold** [comment]
@changtengfei Perfect, I will create a new PR. No, I haven't tested the code yet, I'm reading a little about how OpenWSN works and how to set it up, I have never used it before so this is a first time for me! 

I will create the PR and will let you know as soon as I test the code on the boards!

**Comment by changtengfei** [comment]
This PR is replaced by https://github.com/openwsn-berkeley/openwsn-fw/pull/503, which has been merged.

**Pull request closed without merging**